src/js/components/search/filters/programSource/SourceSelectFilter.jsx

```diff
@@ -0,0 +1,166 @@
+        {
+            code: '020',
+            name: 'Department of the Treasury (TREAS)'
+        },
+        {
+            code: '014',
+            name: 'Department of the Interior (DOI)'
+        },
+        {
+            code: '068',
+            name: 'Environmental Protection Agency (EPA)'
+        },
+        {
+            code: '9876'
+        }
+    ],
+    selectedSources: [],
+    required: false
+};
+
+export class SourceSelectFilter extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            searchString: '',
+            autocompleteOptions: [],
+            noResults: false
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.clearAutocompleteSuggestions = this.clearAutocompleteSuggestions.bind(this);
+        this.selectSourceComponent = this.selectSourceComponent.bind(this);
+        this.timeout = null;
+    }
+
+    parseAutocompleteOptions(options) {
+        const values = [];
+        if (options && options.length > 0) {
+            options.forEach((item) => {
+                values.push({
+                    title: item.code,
+                    subtitle: item.name || '',
+                    data: item
+                });
+            });
+        }
+
+        this.setState({
+            autocompleteOptions: values
+        });
+    }
+
+    queryAutocomplete(input) {
+        this.setState({
+            noResults: false
+        });
+
+        // Only search if input is 1 or more characters
+        if (input.length >= 1) {
+            this.setState({
+                searchString: input
+            });
+
+            let autocompleteOptions = [];
+
+            const matches = this.props.options.filter((source) => (source.code.includes(this.state.searchString)));
+            // Filter out any selectedSources that may be in the result set
+            if (this.props.selectedSources && this.props.selectedSources.length > 0) {
+                autocompleteOptions = differenceWith(matches, this.props.selectedSources, isEqual);
+            }
+            else {
+                autocompleteOptions = matches;
+            }
+
+            this.parseAutocompleteOptions(autocompleteOptions);
+
+            this.setState({
+                noResults: autocompleteOptions.length === 0
+            });
+        }
+    }
+
+    selectSourceComponent(selectedSource) {
+        this.props.updateComponent(this.props.code, selectedSource.code);
+    }
+
+    clearAutocompleteSuggestions() {
+        this.setState({
+            autocompleteOptions: []
+        });
+    }
+
+    handleTextInput(programSourceInput) {
+        // Clear existing sources to ensure user can't select an old or existing one
+        this.setState({
+            autocompleteOptions: []
+        });
+
+        // Grab input, clear any exiting timeout
+        const input = programSourceInput.target.value;
+        window.clearTimeout(this.timeout);
```

Comment:
I discovered via console error that this required using `event.persist()` since synthetic events are pooled - see [this article](https://reactjs.org/docs/events.html#event-pooling)